x-pack/plugins/apm/common/locator.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { LocatorDefinition } from 'src/plugins/share/common';
+import type { SerializableRecord } from '@kbn/utility-types';
+import * as t from 'io-ts';
+import { isRight } from 'fp-ts/Either';
+
+const serviceOverViewTab = t.keyof({
+  transactions: null,
+  metrics: null,
+  logs: null,
+});
+
+export const APM_APP_LOCATOR = 'APM_LOCATOR';
+
+type ServiceOverViewTab = t.TypeOf<typeof serviceOverViewTab>;
+
+export interface ServiceViewLocatorParams extends SerializableRecord {
+  serviceName?: string;
```

Comment:
Could we get types from the routes instead? cc @dgieselaar

Reply:
Instead of declaring the routes used by the locator manually, I generated them from the APM routers definition using the [`getLocatorEnabledRoutes helper` ](https://github.com/elastic/kibana/blob/cbbc8c4202de6d203d544327ef19f4f0ba3799da/x-pack/plugins/apm/public/locator/get_locator_enabled_routes.ts).